test/dns-resolver.cc

```diff
@@ -29,15 +29,9 @@ TEST_F(DomainNameResolveTest, Succeed) {
           typename decltype(e)::template ErrorsFrom<void, std::tuple<>>,
           std::tuple<std::runtime_error>>);
 
-  auto [future, k] = PromisifyForTest(std::move(e));
-
-  k.Start();
-
-  EventLoop::Default().RunUntil(future);
-
   EXPECT_TRUE(
       std::regex_match(
-          future.get(),
+          *e,
```

Comment:
```suggestion
          *std::move(e),
```

Here and everywhere else where we were doing a `PromisifyForTest(std::move(e))` please!